pkg/controller/externalippool/validate.go

```diff
@@ -90,155 +93,62 @@ func (c *ExternalIPPoolController) ValidateExternalIPPool(review *admv1.Admissio
 	}
 }
 
-func validateIPRangesAndSubnetInfo(externalIPPool crdv1beta1.ExternalIPPool, existingExternalIPPools []*crdv1beta1.ExternalIPPool) (string, bool) {
-	subnetInfo := externalIPPool.Spec.SubnetInfo
-	ipRanges := externalIPPool.Spec.IPRanges
-
-	var subnet *netip.Prefix
-	if subnetInfo != nil {
-		gatewayAddr, err := netip.ParseAddr(subnetInfo.Gateway)
-		if err != nil {
-			return fmt.Sprintf("invalid gateway address %s", subnetInfo.Gateway), false
-		}
+func newAdmissionResponseForErr(err error) *admv1.AdmissionResponse {
+	return &admv1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
 
-		if gatewayAddr.Is4() {
-			if subnetInfo.PrefixLength <= 0 || subnetInfo.PrefixLength >= 32 {
-				return fmt.Sprintf("invalid prefixLength %d", subnetInfo.PrefixLength), false
-			}
-		} else {
-			if subnetInfo.PrefixLength <= 0 || subnetInfo.PrefixLength >= 128 {
-				return fmt.Sprintf("invalid prefixLength %d", subnetInfo.PrefixLength), false
-			}
-		}
-		prefix := netip.PrefixFrom(gatewayAddr, int(subnetInfo.PrefixLength)).Masked()
-		subnet = &prefix
+func validateIPRangesAndSubnetInfoForExternalIPPool(externalIPPool *crdv1beta1.ExternalIPPool, existingExternalIPPools []*crdv1beta1.ExternalIPPool) error {
+	ipRanges := externalIPPool.Spec.IPRanges
+	subnetInfo := externalIPPool.Spec.SubnetInfo
+	if err := validation.ValidateIPRangesAndSubnetInfo(subnetInfo, ipRanges); err != nil {
+		return err
 	}
+	return validateNoOverlappingRanges(ipRanges, existingExternalIPPools, externalIPPool.Name)
```

Comment:
it seems that `ipRange` are normalized twice? Should `ValidateIPRangesAndSubnetInfo` return the normalized ranges to avoid this?